dev-helpers/oauth2-redirect.html

```diff
@@ -6,6 +6,14 @@
 <script>
     'use strict';
     function run () {
+        if (!window.opener) {
+            prompt("Run this in devtools on the Swagger UI page to complete auth:", [
+                `swaggerUIRedirectOauth2.auth.code = "${JSON.stringify((new URLSearchParams(location.search)).get('code'))}";`
```

Comment:
`JSON.stringify()` for me adds `"` character which results in an invalid string since the existing `"` characters you added here are immediately closed again.

---

I am wondering why you are doing `JSON.stringify()` in the first place though since (at least for my oauth provider), just getting the code also works:
 ```
`swaggerUIRedirectOauth2.auth.code = "${(new URLSearchParams(location.search)).get('code')}";`,
```

Reply:
Hmmm yeah, you're right, this does add extra `"`s (must have slipped in a previous version of code).

`JSON.stringify` here helps handle a case when `code` has a quote sign or a backslash in it:

```js
JSON.stringify(String.raw`abc"d\ef`));  // -> "abc\"d\\ef"
```

This is a very fringe case to handle but I think we should do it just in case. I'll update the code to remove the hardcoded `"`s instead.